src/Contract/PlutusData.purs

```diff
@@ -107,30 +103,33 @@ import Ctl.Internal.Types.Redeemer
   , redeemerHash
   , unitRedeemer
   ) as Redeemer
-import Data.Either (Either)
+import Data.Bifunctor (lmap)
+import Data.Either (Either, hush)
 import Data.Map (Map)
 import Data.Maybe (Maybe)
+import Effect.Aff.Class (liftAff)
+
+-- | Retrieve the full resolved datum associated to a given datum hash.
+getDatumByHash :: DataHash -> Contract (Maybe Datum)
+getDatumByHash dataHash = do
+  queryHandle <- getQueryHandle
+  liftAff $ join <<< hush <$> queryHandle.getDatumByHash dataHash
 
--- | Get a `PlutusData` given a `DatumHash`.
-getDatumByHash
-  :: forall (r :: Row Type)
-   . DataHash
-  -> Contract r (Maybe Datum.Datum)
-getDatumByHash = wrapContract <<< QueryM.getDatumByHash
+-- | Retrieve full resolved datums associated with given datum hashes.
+-- | The resulting `Map` will only contain datums that have been successfully
+-- | resolved. This function returns `Nothing` in case of an error during
+-- | response processing (bad HTTP code or response parsing error).
+getDatumsByHashes :: Array DataHash -> Contract (Maybe (Map DataHash Datum))
+getDatumsByHashes hashes = do
+  queryHandle <- getQueryHandle
+  liftAff $ hush <$> queryHandle.getDatumsByHashes hashes
 
--- | Get `PlutusData`s given an `Array` of `DataHash`.
--- | This function discards all possible error getting a `DataHash`.
-getDatumsByHashes
-  :: forall (r :: Row Type)
-   . Array DataHash
-  -> Contract r (Map DataHash Datum.Datum)
-getDatumsByHashes = wrapContract <<< QueryM.getDatumsByHashes
+-- | Retrieve full resolved datums associated with given datum hashes.
+-- | The resulting `Map` will only contain datums that have been successfully
```

Comment:
The change originates from this PR: https://github.com/Plutonomicon/cardano-transaction-lib/pull/1235

It's not impossible for Kupo to replicate ODCs behaviour here (we just call getDatumByHash in parallel), and actually it seems blockfrost also can only query by one hash.

@errfrom What was the motivation for the change?

Reply:
Agreed, we should keep ODCs behaviour here.
Initially I thought it would make sense to fail on the first error and that we don't need to explicitly represent the absence of `Datum` in the `Map`, as this can be easily checked by the user.

Reply:
I think we should just drop the plural handle and implement the traversal in Contract